src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (herbivore.hidden === false && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health === 0) {
```

Comment:
The condition to check if the health equals 0 is not sufficient to determine if the animal is dead. The animal's health could be less than 0 after the bite, so the condition should be `herbivore.health <= 0` to ensure that animals with non-positive health are considered dead.

Reply:
fixed